accepted/2024/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,193 @@
+ and then follow-up SDK updates ship updated packages which still belong to the same .NET 8 package group.
+ Here how this will play out with the tooling support described above:
+1. During GA release, the release team will stage the set of RTM packages for the release. (nuget stage --group-id "net8.0" --package-id <package-file-path>)
+2. Then on the day of the release, all the staged packages will be published using `nuget push --group "net8.0"` command.
+   At this point, the group with id `net8.0` will be empty.
+3. Later, as new builds are being prepared for a patch release, a new set of packages will be staged, to be later published to the same group.
+As there can be multiple candidate builds for a release, there is a need for functionality to discard a stage for a given group `nuget delete --stage "net8.0"`.
+This will remove all the staged but not published packages from the net8.0 group.
+4. A new set of packages will be staged for the 8.0 group, until the final build is known. At this point, there is only one unified set of related packages is staged.
+5. On the release date, the release team will call `nuget push --group "net8.0"` again, and only the staged packages will be published, resulting in an expanded set of published packages in the net8.0 group.
+6. Some time in the future, when the release is already out of support, somebody from the .NET team will call
+ `nuget deprecate --stage "net8.0"` and all the packages which have ever been published to that group will be deprecated.
+ Those packages, which has already been deprecated because of whatever reason, will not be altered.
+
+#### Asynchronous behavior
+
+As both publishing and deprecation commands can take some time, these will run asynchronously.
+ So the CLI command for both operations will return immediately, giving the user some token which can later be used to
+ track the current state of the request.
+There are no critical scenarios for deprecation to happen within specific time period, however, publishing is somewhat
+ critical and is being orchestrated in many cases. Hence, NuGet should provide some guarantees for how much time the operation
+ can take in the worst case. As of right now, having 10 minutes deadline for the publishing request should be reasonable.
+
+This requirement brings a need for a new NuGet command for querying the status of an asynchronous operation. This particular requirement should be treated as a stretch goal, and it can be implemented in a later release.
+Let's assume that the group publishing and deprecation commands will return some "operation id" for later state inquriry by the client.
+
+The below command for tracking progress should be treated as a P2 ask, and can be implemented later.
+`nuget status --operation-id` command can be used to track the progress of an asynchronous operation.
+This can return the following results:
+- `success` : when successful, the operation should also return the time when the operation was completed.
```

Comment:
Right now you get an e-mail for package published, should it be the same?

Reply:
I think it can be. But I'll let it up to the implementation to the eng. team to decide how to move forward. Maybe in V1 it's fine to stick with the current model and then only after getting enough requests for this (if any), decide to implement this.

Reply:
I think e-mails are likely to be easier to add/are a non-binding UX compared to a complete new command.